### PR TITLE
8 swerve vision

### PR DESCRIPTION
**Pull request by alt09**
have the pose estimator working  and logging values

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/drive/ModuleIOSparkMax.java [hunk 1]
```diff
@@ -46,25 +46,25 @@ public ModuleIOSparkMax(int index) {
       case 0:
         driveSparkMax = new CANSparkMax(DriveMotor.frontLeft.CAN_ID, MotorType.kBrushless);
         turnSparkMax = new CANSparkMax(TurnMotor.frontLeft.CAN_ID, MotorType.kBrushless);
-        turnAbsoluteEncoder = new CANCoder(8);//TODO: UPDATE CAN IDS
+        turnAbsoluteEncoder = new CANCoder(8);
         absoluteEncoderOffset = -267.6269 + 180;
```
Comment (nancyesquivel10) [inline src/main/java/frc/robot/subsystems/drive/ModuleIOSparkMax.java]: please put this into a constant

File: src/main/java/frc/robot/subsystems/drive/ModuleIOSparkMax.java [hunk 2]
```diff
@@ -46,25 +46,25 @@ public ModuleIOSparkMax(int index) {
...
         break;
       case 1:
         driveSparkMax = new CANSparkMax(DriveMotor.frontRight.CAN_ID, MotorType.kBrushless);
         turnSparkMax = new CANSparkMax(TurnMotor.frontRight.CAN_ID, MotorType.kBrushless);
-        turnAbsoluteEncoder = new CANCoder(11);//TODO: UPDATE CAN IDS
+        turnAbsoluteEncoder = new CANCoder(11);
         absoluteEncoderOffset = -269.7363 + 180;
```
Comment (nancyesquivel10) [inline src/main/java/frc/robot/subsystems/drive/ModuleIOSparkMax.java]: same here constants

File: src/main/java/frc/robot/subsystems/drive/ModuleIOSparkMax.java [hunk 3]
```diff
@@ -46,25 +46,25 @@ public ModuleIOSparkMax(int index) {
...
         break;
       case 2:
         driveSparkMax = new CANSparkMax(DriveMotor.backLeft.CAN_ID, MotorType.kBrushless);
         turnSparkMax = new CANSparkMax(TurnMotor.backLeft.CAN_ID, MotorType.kBrushless);
-        turnAbsoluteEncoder = new CANCoder(5);//TODO: UPDATE CAN IDS
+        turnAbsoluteEncoder = new CANCoder(5);
         absoluteEncoderOffset = -356.0449;
```
Comment (nancyesquivel10) [inline src/main/java/frc/robot/subsystems/drive/ModuleIOSparkMax.java]: add to constants

File: src/main/java/frc/robot/subsystems/drive/ModuleIOSparkMax.java [hunk 4]
```diff
@@ -46,25 +46,25 @@ public ModuleIOSparkMax(int index) {
...
         break;
       case 3:
         driveSparkMax = new CANSparkMax(DriveMotor.backRight.CAN_ID, MotorType.kBrushless);
         turnSparkMax = new CANSparkMax(TurnMotor.backRight.CAN_ID, MotorType.kBrushless);
-        turnAbsoluteEncoder = new CANCoder(2);//TODO: UPDATE CAN IDS
+        turnAbsoluteEncoder = new CANCoder(2);
         absoluteEncoderOffset = -177.6269;
```
Comment (nancyesquivel10) [inline src/main/java/frc/robot/subsystems/drive/ModuleIOSparkMax.java]: add to constants

File: src/main/java/frc/robot/subsystems/pose/Pose.java [hunk 5]
```diff
@@ -80,39 +84,45 @@ public void periodic() {
...
-            if (target.getPoseAmbiguity() <= 0.2 && fiducialid >= 0 && fiducialid < 9) {
-
+            if (target.getPoseAmbiguity() >= 0.2 && fiducialid >= 0 && fiducialid < 9) {
+                
                 AprilTagFieldLayout atfl;
                 try {
                     atfl = new AprilTagFieldLayout(AprilTagFields.k2023ChargedUp.loadAprilTagLayoutField().getTags(),
-                            16.4592,
-                            8.2296);
+                    16.4592,
```
Comment (nancyesquivel10) [inline src/main/java/frc/robot/subsystems/pose/Pose.java]: put this field info in constants please

File: src/main/java/frc/robot/subsystems/pose/Pose.java [hunk 6]
```diff
@@ -80,39 +84,45 @@ public void periodic() {
...
+                    8.2296);
```
Comment (nancyesquivel10) [inline src/main/java/frc/robot/subsystems/pose/Pose.java]: put this field info in constants please

**Review by nancyesquivel10 — COMMENTED** [review]
add these values in constants

**Review by nancyesquivel10 — DISMISSED** [review]
lgtm

File: src/main/java/frc/robot/Robot.java [hunk 7]
```diff
@@ -24,6 +29,8 @@
 public class Robot extends LoggedRobot {
   private RobotContainer robotContainer = new RobotContainer();
   private Command m_AutoCommand;
+  private VisionIOArduCam m_ArduCam;
```
Comment (El1zabethS) [inline src/main/java/frc/robot/Robot.java]: Why are lines 32 and 33 there? Are they necessary? Remove if not.

**Review by El1zabethS — CHANGES_REQUESTED** [review]
Look at comments

**Review by El1zabethS — APPROVED** [review]

**Review by nancyesquivel10 — APPROVED** [review]

**Pull request merged**